libsplinter/src/circuit/template/mod.rs

```diff
@@ -38,38 +40,46 @@ pub(self) use crate::admin::messages::{CreateCircuitBuilder, SplinterServiceBuil
 
 /// Default file location for circuit templates
 pub const DEFAULT_TEMPLATE_DIR: &str = "/usr/share/splinter/circuit-templates";
+/// Environment variable for file location for circuit templates
+pub const SPLINTER_CIRCUIT_TEMPLATE_PATH: &str = "SPLINTER_CIRCUIT_TEMPLATE_PATH";
 
 /// Manages circuit templates.
 ///
 /// `CircuitTemplateManager` maintains the location of circuit templates, and may be used to
 /// list any availabe circuit templates found in the `path` of the `CircuitTemplateManager`.
 pub struct CircuitTemplateManager {
     /// Path of the directory containing the circuit template files.
-    path: String,
+    paths: Vec<String>,
 }
 
 impl Default for CircuitTemplateManager {
     /// Constructs a `CircuitTemplateManager` with the `DEFAULT_TEMPLATE_DIR`.
     fn default() -> Self {
         CircuitTemplateManager {
-            path: DEFAULT_TEMPLATE_DIR.to_string(),
+            paths: vec![DEFAULT_TEMPLATE_DIR.to_string()],
         }
     }
 }
 
 impl CircuitTemplateManager {
     /// Constructs a `CircuitTemplateManager` with a custom `path` to the circuit templates.
-    pub fn new(path: &str) -> Result<CircuitTemplateManager, CircuitTemplateError> {
-        if !Path::new(&path).is_dir() {
-            Err(CircuitTemplateError::new(&format!(
-                "{} is not a valid directory",
-                path
-            )))
-        } else {
-            Ok(CircuitTemplateManager {
-                path: path.to_string(),
+    pub fn new(paths: &[String]) -> CircuitTemplateManager {
+        let mut paths: Vec<String> = paths
+            .iter()
+            .filter_map(|path| {
+                if Path::new(&path).is_dir() {
+                    Some(path.to_string())
+                } else {
+                    None
+                }
             })
+            .collect();
+        if let Ok(template_paths) = env::var(SPLINTER_CIRCUIT_TEMPLATE_PATH) {
```

Comment:
This should be pushed up a level outside CircuitTemplateManager; then paths passed in wouldn't need to be modified and the caller of new() is in full control of the behavior.

Reply:
I think it would make sense for part of this check to get moved into the `find_template` method, in the case that the function does not have any paths to search. Then, the `new` method is in full control of the caller (unless they don't provide any paths, in which case it'll still search the env var paths if set and the default)